packages/app/features/top/UserProfileScreen.tsx

```diff
@@ -90,8 +90,11 @@ export function UserProfileScreen({ route, navigation }: Props) {
 
   const shouldShowBackButton = useMemo(() => {
     const isWebDesktop = isWeb && !isWindowNarrow;
-    const isContactsTabRoot = navigation.getState().history?.length === 1;
-    return !(isWebDesktop && isContactsTabRoot);
+    const navHistory = navigation.getState().history;
+    const isContactsTabRoot = navHistory?.length === 1;
+    // @ts-expect-error - key is a valid property
+    const isActivityTab = navHistory?.[0]?.key.includes('ActivityEmpty');
+    return !((isWebDesktop && isContactsTabRoot) || isActivityTab);
```

Comment:
we want the same `isWebDesktop` check for the `isActivityTab` conditional, right? On mobile, we need that back button